Cython/Compiler/ParseTreeTransforms.py

```diff
@@ -1366,14 +1366,19 @@ def visit_ComprehensionNode(self, node):
                 itseq.function.obj = obj
                 node = EvalWithTempExprNode(obj, node)
 
-            for n, a in enumerate(itseq.args):
+            for n, a in reversed(list(enumerate(itseq.args))):
+                # "reversed" preserved the order the arguments are evaluated
                 if isinstance(a, ExprNodes.ConstNode):
                     continue
                 a = ResultRefNode(a)
                 itseq.args[n] = a
```

Comment:
It seems to me that this is does nothing useful and might be artifact from older version.  Since `itseq` is replace by `new_*` (otherwise after this loop half of the attrs will be lost), can we just do `for a in reversed(itseq.args)`?

Reply:
I'm pretty sure this is right. I use `n` to index `itseq.args`. `itseq` still exists and is valid. `new_itseq` is just a `ResultRefNode` that contains `itseq`

Reply:
> `new_itseq` is just a `ResultRefNode` that contains `itseq`

I'm so sorry, I did not know about that.  I am wondering tho, is it necessary to unlazy `itseq.args` (`list`) then?

Reply:
I'm not sure I understand the question fully, but I don't think so. The reference to `itseq` is shared between `itseq` and the `ResultRefNode` so changing one changes both.

Reply:
I was having some confusion earlier interpreting the `reversed(list(enumerate` call.  My latest referred to that can we just `reversed(enumerate` which is lazier and may reduce some tiny overhead.  Please the discard my embarrassing *suggest changes* thing if it is possible though.